### PR TITLE
Adding new statuses and a property to EmailDeliveryReportReceived Event

### PR DESCRIPTION
**Pull request by anmolbohra97**
# Data Plane API - Pull Request
Adding new statuses "Bounced" and "Suppressed" to the EmailDeliveryReportReceivedEvent status and also adding a new property deliveryStatusDetails

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [ ] a public preview
- [ ] GA release

### Change Scope
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document:
* Previous API Spec Doc:
* Updated paths:

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links
 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

### CODE REVIEW

**Review by JoshLove-msft — APPROVED** [review]

**Review by kagbakpem — CHANGES_REQUESTED** [review]
Hold off on this PR. Current internal discussion might trigger additional changes.

**Comment by JoshLove-msft** [comment]
> Hold off on this PR. Current internal discussion might trigger additional changes.

Would you like us to review these changes at the arch board next week?

**Review by kagbakpem — APPROVED** [review]

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json
Comment (JoshLove-msft) [inline specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:537]: We need an example that shows this.

Comment (JoshLove-msft) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:537]: Potentially make this an object so it can evolve over time, and the string information can go in a `Message` or `Description` property.

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json
Comment (JoshLove-msft) [inline specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:522]: Update to mention that any value other than "Delivered" is a failure to deliver.

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json
Comment (JoshLove-msft) [inline specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:529]: Consider adding descriptions for each value - https://azure.github.io/autorest/extensions/#x-ms-enum

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json
Comment (anmolbohra97) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:564]: Updated this property to be an object with 'statusDescription' as one of the properties.

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json
Comment (anmolbohra97) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:529]: Added descriptions for enum values.

**Review by JoshLove-msft — APPROVED** [review]
@lmazuel can you please merge?

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json
Comment (JoshLove-msft) [inline specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:563]: Actually, now that this is an object, you'll need to define it as a separate definition. I'd suggest calling it AcsEmailDeliveryStatusDetails.

**Comment by JoshLove-msft** [comment]
> @lmazuel can you please merge?

Actually, one more comment that I just added needs to be addressed before merging.

**Review by JoshLove-msft — CHANGES_REQUESTED** [review]
Will need to add a new definition for the delivery status details,.

File: specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json
Comment (anmolbohra97) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:563]: I see other objects being defined in place. Any particular reason why I need to define it separately?

Comment (JoshLove-msft) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:563]: There shouldn't be other objects like this other than ones where additionalProperties is set to true. This will not work with the code gen as it doesn't know what to name the type.

Comment (anmolbohra97) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:563]: Actually, nevermind. Just saw the definition

Comment (anmolbohra97) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:563]: Updating

Comment (lmazuel) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json:563]: SDK needs to generate a class, and anonymous models forces the SDK generator to come up with a name by itself if it wants to generate. For this reason, this pattern is now forbideen, and codegen shall fail until you give an actual name.

Comment (JoshLove-msft) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json]: Can we actually call this AcsEmailDeliveryReportStatusDetails instead? Sorry for not suggesting this initially!

Comment (anmolbohra97) [reply specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json]: Sure. Updated.

**Review by JoshLove-msft — APPROVED** [review]

**Review by kagbakpem — APPROVED** [review]

**Comment by anmolbohra97** [comment]
@lmazuel @JoshLove-msft can this now be merged?

**Comment by JoshLove-msft** [comment]
> @lmazuel @JoshLove-msft can this now be merged?

Yes, @lmazuel can you please merge?

**Review by lmazuel — APPROVED** [review]

**Pull request merged**
